subprojects/packagefiles/curl/meson.build

```diff
@@ -0,0 +1,1866 @@
+# Init. {{{
+
+project(
+  'curl',
+  'c',
+  version: '8.0.1',
+  # Version gated use of `test(…, verbose: true)` needs `0.62.0`.
+  meson_version: meson.version().version_compare('>=0.62.0') ? '>=0.62.0' : '>=0.60.0',
+)
+
+fs = import('fs')
+pkg = import('pkgconfig')
+cc = meson.get_compiler('c')
+cdata = configuration_data()
+
+sys_deps = []
+lib_deps = []
+
+# Constants. {{{
+
+CURL_PUBLIC_INC = include_directories('include')
+CURL_INTERNAL_INC = include_directories('lib')
+CURL_VERSION = cc.get_define(
+  'LIBCURL_VERSION',
+  include_directories: CURL_PUBLIC_INC,
+  prefix: '#include"curl/curlver.h"',
+).strip('"')
+CURL_VERSION_NUM = cc.get_define(
+  'LIBCURL_VERSION_NUM',
+  include_directories: CURL_PUBLIC_INC,
+  prefix: '#include"curl/curlver.h"',
+)
+
+if CURL_VERSION != meson.project_version()
+  error(
+    'versions mismatch, meson has @0@, while `culver.h` reports @1@!'.format(
+      meson.project_version(),
+      CURL_VERSION,
+    ),
+  )
+endif
+
+WINDOWS_ONLY_ERROR = 'only supported on Windows'
+MACOS_ONLY_ERROR   = 'only supported on macOS'
+
+MISSING_LIBS_ERROR = 'required libraries missing'
+
+SSL_DISABLED_ERROR  = 'feature ssl is disabled'
+HTTP_DISABLED_ERROR = 'feature http is disabled'
+
+DISABLED_OPT = get_option('_disabled')
+ENABLED_OPT  = get_option('_enabled')
+
+# }}}
+
+# Basic feature options. {{{
+# (not dependent on checks)
+
+foreach _opt : [
+  'cookies',
+  'crypto-auth',
+  'dict',
+  'doh',
+  'file',
+  'ftp',
+  'getoptions',
+  'gopher',
+  'imap',
+  'libcurl-option',
+  'mime',
+  'mqtt',
+  'netrc',
+  'parsedate',
+  'pop3',
+  'progress-meter',
+  'proxy',
+  'rtsp',
+  'shuffle-dns',
+  'smtp',
+  'socketpair',
+  'telnet',
+  'tftp',
+  'verbose-strings',
+]
+  set_variable(_opt.underscorify() + '_opt', get_option(_opt))
+endforeach
+
+# }}}
+
+# }}}
+
+# Platform dependencies. {{{
+
+if host_machine.system() == 'windows'
+  # For `BCryptGenRandom()`.
+  sys_deps += cc.find_library('bcrypt', required: false)
+  winmm_dep = cc.find_library('winmm', required: false)
+  if winmm_dep.found() and cc.has_function('getch', dependencies: winmm_dep)
+    sys_deps += winmm_dep
+  endif
+  ws2_32_dep = cc.find_library('ws2_32', required: false)
+  if ws2_32_dep.found() and cc.has_function('getch', dependencies: ws2_32_dep)
+    sys_deps += ws2_32_dep
+  endif
+endif
+
+# For `clock_gettime`.
+sys_deps += cc.find_library('rt', required: false)
+
+if host_machine.system() == 'darwin'
+  sys_deps += dependency(
+    'appleframeworks',
+    modules: [
+      'CoreFoundation',
+      'SystemConfiguration',
+    ],
+  )
+endif
+
+threads_dep = dependency('threads', required: false)
+
+# }}}
+
+# Basic checks. {{{
+
+check_args = []
+check_headers = []
+
+_headerlist = {
+  'arpa/inet.h': [
+    'getpeername',
```

Comment:
that is so hard to read. Maybe put the array on the same line.

Reply:
It's easier as soon as they are multiple functions, and `muon fmt` will automatically switch to multiple-lines in that case anyway. I try to limit my fighting with its choices.